populator/nosql/couchbase/src/main/java/org/jboss/arquillian/populator/nosql/couchbase/CouchbaseOptions.java

```diff
@@ -0,0 +1,112 @@
+package org.jboss.arquillian.populator.nosql.couchbase;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Class taht represents custom options for Couchbase
+ */
+public class CouchbaseOptions implements Map<String, Object> {
+
+    static String BUCKET_PASSWORD = "BUCKET_PASSWORD";
+    static String CREATE_BUCKET = "CREATE_BUCKET";
+    static String CLUSTER_USERNAME = "CLUSTER_USERNAME";
+    static String CLUSTER_PASSWORD = "CLUSTER_PASSWORD";
+
+    private Map<String, Object> options = new HashMap<>();
+
+    private CouchbaseOptions() {
+    }
+
+    public static CouchbaseConfigurationOptions options() {
+        return new CouchbaseConfigurationOptions();
+    }
+
+    @Override
+    public int size() {
+        return options.size();
+    }
+
+    @Override
+    public boolean isEmpty() {
```

Comment:
If we encapsulate underlying Map anyway maybe we can think of better names for those delegate methods? e.g. `options.amount()` and `options.contains()`. Key reveals underlying implementation to some extent but also does not really bring any extra meaning.